packages/react-shopify-app-route-propagator/README.md

```diff
@@ -65,74 +30,135 @@ export interface Props {
 }
 ```
 
-Allowing you to use it with a string
+### `useRoutePropagation`
 
-```javascript
-<ShopifyRoutePropagator location="/foo/bar?thing=true" app={app} />
-```
-
-Or a location object
+This example uses [app bridge](https://help.shopify.com/en/api/embedded-apps/app-bridge#set-up-your-app) to create an app instance and the `withRouter` enhancer from `react-router-router`.
 
 ```javascript
-<ShopifyRoutePropagator location={this.props.location} app={app} />
-```
+import React from 'react';
+import {Switch, Route, withRouter} from 'react-router'
+import {BrowserRouter} from 'react-router-dom'
+import createApp, {getShopOrigin} from '@shopify/app-bridge';
+import {useRoutePropagation} from '@shopify/react-shopify-app-route-propagator';
 
-Or even with `window.location` (though then you will need to make sure it rerenders yourself)
+export default withRouter(function(props) {
+  useRoutePropagation(
+    createApp({
```

Comment:
maybe, maybe not? but since both polaris & app bridge does not have context API or hook support, I really don't know where the app could be coming from.

I think the safest thing is to assume you create it here, and that the consumer will read the app bridge documentation and figure out they need to find a way to share this app instance throughout out the entire app.

Reply:
Still feels weird to me, I would probably make a note of it (I think they could certainly write their own decorator to get the app bridge app out of context).